common/nunjucks/README.md

```diff
@@ -0,0 +1,114 @@
+# @curiostack/nunjucks
+
+An extended version of [mozilla/nunjucks](https://github.com/mozilla/nunjucks),
+with some extra convenience methods added on.
+
+## Usage
+
+### Import
+
+The extended `nunjucks` object can be imported in other `curiostack` `yarn` modules
+using the package name `@curiostack/nunjucks`.
+
+```typescript
+import nunjucks from '@curiostack/nunjucks'
+// no need for require('nunjucks')
+
+const env = nunjucks.Environment()
+// ...
+```
+
+### API
+
+The underlying API of `nunjucks` is unchanged, so you can
+[use the official docs](https://mozilla.github.io/nunjucks/getting-started.html) for reference.
+
+## Environment
+
+The [nunjucks Environment](https://mozilla.github.io/nunjucks/api.html#environment)
+class still functions as before. It just has been extended with some extra functionality.
+
+```typescript
+import nunjucks from '@curiostack/nunjucks'
+
+const env = nunjucks.Environment()
+// ...
+
+const result = env.render('template.njk')
+// `result` is rendered with all of the base `nunjucks.Environment` logic
+//   as well as the extended `curiostack` `nunjucks.Environment` functionality
+```
+
+> This means that you can NOT access the extended features, if you use the
+> [simple nunjucks API](https://mozilla.github.io/nunjucks/api.html#simple-api),
+> without initializing the extended `nunjucks.Environment`
+
+### Filters
+
+The `Environment` object is extended with some extra convenience filters.
+
+#### allAfterTaggedLine(tag: string, [useRegex: boolean])
+
+Scans the content of a multiline text and returns all the lines after a tagged line.
+
+A tagged line is a line which contains the `tag` text within it.
```

Comment:
One of my intention is to use `allAfter` in cases where I want to include an example class, but not include metadata which isn't important for the docs.

Something like:
```java
package dont.care

import dont.care.Class1
import dont.care.Class2
import dont.care.Class3

// tag here, which is used by `allAfter` filter
public class WantThis {
  // .. some actually interesting code
}
```

Reply:
> Let's make the default an exact match, it's more natural to use a regex to make things more general rather than use it to make things more precise, e.g. adding `^$`

This one I'll definitely do. I was split on which one to make the default, so your suggestion is a good tiebreaker.